src/components/calendar/drag-drop-provider.tsx

```
@@ -0,0 +1,353 @@
+  const { scrollBy, onDragUpdate } = useScrolling(
+    listRef,
+    currentScrollOffsetRef,
+    viewportHeightRef
+  );
+  const { computeTargetDate, completeDrop } = useDropCompletion(
+    draggedTaskRef,
+    undoRef,
+    performUndo,
+    (isDragging: boolean) => {
+      if (!isDragging) {
+        draggedTaskRef.current = undefined;
```

ESLint

[error] Mutating a value returned from a function whose return value should not be mutated

(react-compiler/react-compiler)